lib/mimic/server.ex

```diff
@@ -441,6 +441,11 @@ defmodule Mimic.Server do
         state.reset_tasks
       end
 
+    # Clear the beam modules after starting the tasks (they read the state)
```

Comment:
I've cherry-picked this commit and pushed to `main` as this is obviously the right thing to do :+1:

Reply:
Sounds good. The other ones should be resolved by the fix in Elixir.